bokehjs/src/coffee/models/annotations/whisker.coffee

```diff
@@ -0,0 +1,119 @@
+import {Annotation, AnnotationView} from "./annotation"
+import {ColumnDataSource} from "../sources/column_data_source"
+
+import * as p from "core/properties"
+
+export class WhiskerView extends AnnotationView
+  initialize: (options) ->
+    super(options)
+    @set_data(@model.source)
+
+  connect_signals: () ->
+    super()
+    @connect(@model.source.streaming, () -> @set_data(@model.source))
+    @connect(@model.source.patching, () -> @set_data(@model.source))
+    @connect(@model.source.change, () -> @set_data(@model.source))
+
+  set_data: (source) ->
+    super(source)
+    @visuals.warm_cache(source)
+    @plot_view.request_render()
+
+  _map_data: () ->
+    x_scale = @plot_view.frame.xscales[@model.x_range_name]
+    y_scale = @plot_view.frame.yscales[@model.y_range_name]
+
+    limit_scale = if @model.dimension == "height" then y_scale else x_scale
+    base_scale = if @model.dimension == "height" then x_scale else y_scale
+
+    if @model.lower.units == "data"
+      _lower_vx = limit_scale.v_compute(@_lower)
+    else
+      _lower_vx = @_lower
+
+    if @model.upper.units == "data"
+      _upper_vx = limit_scale.v_compute(@_upper)
+    else
+      _upper_vx = @_upper
+
+    if @model.base.units == "data"
+      _base_vx = base_scale.v_compute(@_base)
```

Comment:
Never mind, `base` is a spec, so `_base` is magically created in `materialize_dataspecs()`.